src/libmongoc/src/mongoc/mongoc-client-pool.c

```diff
@@ -117,6 +117,7 @@ mongoc_client_pool_new (const mongoc_uri_t *uri)
 
    pool = (mongoc_client_pool_t *) bson_malloc0 (sizeof *pool);
    bson_mutex_init (&pool->mutex);
+   mongoc_cond_init (&pool->cond);
```

Comment:
If you removed (or commented out) this line would your test still pass locally? For me it does so I'm wondering if there's something I'm doing wrong on my part. I also wouldn't be opposed to not adding a test since we're just initializing a member variable. @kevinAlbs let me know if you disagree.

Reply:
It fails for me when I comment out that line.  I believe it is undefined behavior and might fail or might not (and maybe always works on some systems).  There is also some non-determinism involved.  I would be ok with not adding the test.  I added it because I don't think we have coverage for this kind of case, where one thread is waiting for another thread to push a client to the pool.

Reply:
It fails on my mac. It does not surprise me that this does not fail on Linux. Popping more clients than available in `maxPoolSize` seems like a common thing to do. If this does not cause any issues on Linux, that could explain why we have not had any reports about this.

I am in favor of the test. Even though the fix is simple, this should help prevent future bugs if we change the client pool behavior.